examples/chat/module/rust/src/lib.rs

```diff
@@ -30,6 +30,29 @@ mod proto {
     include!(concat!(env!("OUT_DIR"), "/oak.examples.chat.rs"));
 }
 
+oak::entrypoint_command_handler!(main => Main);
+
+/// Main entrypoint of the chat application.
+///
+/// This node is in charge of creating the other top-level nodes, but does not process any request.
+struct Main;
+
+impl oak::CommandHandler<ConfigMap> for Main {
+    fn handle_command(&mut self, _command: ConfigMap) -> anyhow::Result<()> {
+        let grpc_channel =
+            oak::grpc::server::init("[::]:8080").expect("could not create gRPC server pseudo-Node");
+        oak::node_create(
+            &oak::node_config::wasm("app", "router"),
+            &Label::public_untrusted(),
+            grpc_channel.handle,
+        )
+        .expect("could not create router node");
+        Ok(())
+    }
+}
+
+oak::entrypoint_command_handler!(router => Router::default());
```

Comment:
What if someone would want to initialize Router - for example pass parameters with gRPC address to which to connect through a gRPC client?

Reply:
I am working on that, anyways that's an orthogonal issue, right? That's why we are having the discussion on using `Either` or a tuple.

Reply:
If so, them why is separate `Main` node necessary, could Router just be the main node?

Reply:
That's what I am trying to avoid. I would like each node to correspond to a loop that only handles one type of commands, to keep things organized. This has the advantage that the main node only sets up other nodes, and does not try to read from other channel at some point. Also it means that the main node may be replaced by something more declarative based on a config if we want.

Reply:
So right now I need to send init from `Main` to `Router`, right?

Reply:
Yes, using this patter we would have to do that.

Reply:
Also, since we provide `Router` with a gRPC server channel, how can we send an `init` message to it? Since gRPC channel already has a defined message type.

Reply:
I am hoping Algebraic Data Types, specifically `Either` (or `Pair`), would help here. Haven't fully thought it out though.

Reply:
I don't think `Either` would help here.
`Main` creates both `gRPC` and `Router` nodes, and while creating `Router` it provides it with `gRPC` channel (so `gRPC` and `Router` are connected). But the `init` message should be sent from `Main` to `Router`, so it would need an additional channel for this.

Reply:
You are right, actually I think that @wildarch 's suggestion would work better here: https://github.com/project-oak/oak/issues/1639#issuecomment-718848474